### ISSUE TITLE
Epson Perfection V39 II Scanner Support Planned?

### ISSUE THREAD

**Issue by Nudiventra** [brlin-tw/iscan-snap#11]
I purchased this scanner without checking Linux compatibility. I’m used to everything working by default with my old HP, and I naively assumed it would be the same now. I've tried everything; the computer recognizes the scanner, but without SANE support, it seems pointless:

[ 1181.624958] usb 3-9: Product: Epson Perfection V39II
[ 1181.624962] usb 3-9: Manufacturer: EPSON
[ 1181.624965] usb 3-9: SerialNumber: XBZZ052402

Here, if you search for the V39 model, they mention that the driver also supports V39I...
[](https://download.ebz.epson.net/dsc/search/01/search/searchModule)
I installed it, but nothing works!

Is there any solution? Is support for this model planned? Thank you!

**Comment by brlin-tw** [comment]
Hello, as I personally don't own a Perfection V39 II I don't know whether it can be supported as well.

You probably need to first investigate whether the product has a corresponding non-free plugin package like the following URLs as it likely needs to load a non-free firmware to function in the first place:

https://github.com/brlin-tw/iscan-snap/blob/master/snap/local/utilities/install-non-free-plugins.sh#L13-L30

Good luck!

**Comment by brlin-tw** [comment]
> Here, if you search for the V39 model, they mention that the driver also supports V39II

You may want to try the following instructions:

1. Launch a text terminal.
2. Run the `iscan.install-nonfree-plugins` command to launch the "Install non-free EPSON scanner plugins" utility and install the Perfection V19/V39(gt-s650) non-free plugin.
3. Connect the scanner to the system.
4. Run the `lsusb` command to enumerate connected USB devices, then take note of the Bus# and Device# of the scanner device(including the leading zeros).
5. Run the following commands to temporarily grant scanner device write access to everyone on the system:

    ```bash
    bus_no=_enter_bus_no_here_
    device_no=_enter_device_no_here_
    device_node="/dev/bus/usb/${bus_no}/${device_no}"
    if ! sudo chmod --verbose a+x "${device_node}"; then
        printf 'Error: Unable to grant write permission to the scanner device.\n' 1>&2
    else
        printf 'Info: Operation successful.\n'
    fi
    ```

6. Run the following commands to ensure the snap can access the scanner device:

    ```bash
    sudo snap connect iscan:raw-usb
    ```

7. Run the following command to launch the application and verify whether it is working:

    ```bash
    snap run iscan
    ```

**Comment by Nudiventra** [comment]
> > Here, if you search for the V39 model, they mention that the driver also supports V39II
> 
> You may want to try the following instructions:
> 
> 1. Launch a text terminal.
> 2. Run the `iscan.install-nonfree-plugins` command to launch the "Install non-free EPSON scanner plugins" utility and install the Perfection V19/V39(gt-s650) non-free plugin.
> 3. Connect the scanner to the system.
> 4. Run the `lsusb` command to enumerate connected USB devices, then take note of the Bus# and Device# of the scanner device(including the leading zeros).
> 5. Run the following commands to temporarily grant scanner device write access to everyone on the system:
>    ```shell
>    bus_no=_enter_bus_no_here_
>    device_no=_enter_device_no_here_
>    device_node="/dev/bus/usb/${bus_no}/${device_no}"
>    if ! sudo chmod --verbose a+x "${device_node}"; then
>        printf 'Error: Unable to grant write permission to the scanner device.\n' 1>&2
>    else
>        printf 'Info: Operation successful.\n'
>    fi
>    ```
> 6. Run the following commands to ensure the snap can access the scanner device:
>    ```shell
>    sudo snap connect iscan:raw-usb
>    ```
> 7. Run the following command to launch the application and verify whether it is working:
>    ```shell
>    snap run iscan
>    ```

Following your instructions was successful, but unfortunately when I run iscan I get the message "Failed to send command to scanner. Check scanner status". I tried using neural network to troubleshoot, but we just go in circles: "Based on the current results, saned.socket is still down and saned.service remains hidden. Let's try to take some steps to find out". And so on, over and over.

C:\home\igor> lsusb
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 001 Device 002: ID 8087:800a Intel Corp. Hub
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 002 Device 002: ID 8087:8002 Intel Corp. 8 channel internal hub
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 003 Device 002: ID 17ef:6123 Lenovo USBC Wired OpticalMouse
Bus 003 Device 003: ID 0c76:8888 JMTek, LLC. ME6S
Bus 003 Device 004: ID 09da:2268 A4Tech Co., Ltd. USB Keyboard
Bus 003 Device 007: ID 0480:0901 Toshiba America Inc EXTERNAL_USB
Bus 003 Device 008: ID 045e:028e Microsoft Corp. Xbox360 Controller
Bus 003 Device 012: ID 04b8:013f Seiko Epson Corp. Epson Perfection V39II
Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
C:\home\igor> bus_no=003
device_no=012
device_node="/dev/bus/usb/${bus_no}/${device_no}"

if ! sudo chmod --verbose a+x "${device_node}"; then
    printf 'Error: Unable to grant write permission to the scanner device.\n' 1>&2
else
    printf 'Info: Operation successful.\n'
fi
[sudo] пароль для igor: 
права доступа '/dev/bus/usb/003/012' изменены с 0666 (rw-rw-rw-) на 0777 (rwxrwxrwx)
Info: Operation successful.
C:\home\igor> sudo snap connect iscan:raw-usb
C:\home\igor> snap run iscan


Could not send command to scanner. Checkthe scanner's status.


In short, I give up! I've already resigned myself to the fact that it won't work, and I'll have to install Windows somewhere separately to use the scanner. Thank you for your help!

**Comment by Nudiventra** [comment]
I forgot to mention this earlier. The problem has been resolved! On the https://download.ebz.epson.net/dsc/search/01/search/?OSC=LX, you can find a driver for this scanner along with the Epson Scan 2 software. The device works only with this program, but it works perfectly.

The tricky part is that the driver is located in an unexpected place. If you search for Epson Perfection V39II, you won't find a Linux driver. However, it's actually available if you search for the Linux driver for Epson Perfection V39 (the previous model).

I hope this information helps someone!

The scanner itself is truly excellent!
